core/execution/amm/pool.go

```diff
@@ -488,57 +508,150 @@ func impliedPosition(sqrtPrice, sqrtHigh num.Decimal, l num.Decimal) *num.Uint {
 
 // OrderbookShape returns slices of virtual buy and sell orders that the AMM has over a given range
 // and is essentially a view on the AMM's personal order-book.
-func (p *Pool) OrderbookShape(from, to *num.Uint) ([]*types.Order, []*types.Order) {
-	buys := []*types.Order{}
-	sells := []*types.Order{}
+func (p *Pool) OrderbookShape(from, to *num.Uint, idgen *idgeneration.IDGenerator) ([]*types.Order, []*types.Order) {
+	buys, sells := []*types.Order{}, []*types.Order{}
 
-	if from == nil {
-		from = p.lower.low
+	lower := p.lower.low
+	upper := p.upper.high
+	fairPrice := p.fairPrice()
+
+	if p.closing() {
+		// AMM is in reduce only mode so will only have orders between its fair-price and its base so shrink from/to to that region
+		pos := p.getPosition()
+		if pos == 0 {
+			// pool is closed and we're waiting for the next MTM to close, so it has no orders
+			return nil, nil
+		}
+
+		if pos > 0 {
+			// only orders between fair-price -> base
+			lower = fairPrice.Clone()
+			upper = p.lower.high.Clone()
+		} else {
+			// only orders between base -> fair-price
+			upper = fairPrice.Clone()
+			lower = p.lower.high.Clone()
+		}
 	}
-	if to == nil {
-		to = p.upper.high
+
+	if from.GT(upper) || to.LT(lower) {
+		return nil, nil
 	}
 
-	// any volume strictly below the fair price will be a buy, and volume above will be a sell
 	side := types.SideBuy
-	fairPrice := p.fairPrice()
+
+	// cap the range to the pool's bounds, there will be no orders outside of this
+	from = num.Max(from, lower)
+	to = num.Min(to, upper)
+
+	switch {
+	case from.GT(fairPrice):
+		// if we are expanding entirely in the sell range to calculate the order at price `from`
+		// we need to ask the AMM for volume in the range `from - 1 -> from` so we simply
+		// sub one here to cover than.
+		side = types.SideSell
+		from.Sub(from, p.oneTick)
+	case to.LT(fairPrice):
+		// if we are expanding entirely in the buy range to calculate the order at price `to`
+		// we need to ask the AMM for volume in the range `to -> to + 1` so we simply
+		// add one here to cover than.
+		to.Add(to, p.oneTick)
+	case from.GTE(fairPrice):
```

Comment:
This feels ambiguous. The first case is `from.GT(fairPrice)`, so realistically, the third case could either be `default:`, or explicitly `from.EQ(fairPrice)`. the `GTE` kind of covers the first case for no good reason